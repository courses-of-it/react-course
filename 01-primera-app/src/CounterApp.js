import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value = 0 }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };
  const handleSubstract = () => {
    setCounter(counter - 1);
  };
  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={ handleAdd }>+</button>
      <button onClick={ handleSubstract }>-</button>
      <button onClick={ handleReset }>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
