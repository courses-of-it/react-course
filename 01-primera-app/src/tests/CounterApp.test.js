import "@testing-library/jest-dom";

import React from "react";
import CounterApp from "../CounterApp";
import { shallow } from "enzyme";

describe("tests to <CounterApp />", () => {
  let wrapper = shallow(<CounterApp value={ 10 } />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={ 10 } />);
  });

  test("mirar si carga el componente correactamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Comprobar si el valor por defecto se asigna correctamente", () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const valueDiv = wrapper.find("h2").text().trim();
    expect(valueDiv).toBe(value.toString());
  });

  test("deve increcmentar el contador", () => {
    wrapper.find("button").at(0).simulate("click");
    const valueDiv = wrapper.find("h2").text().trim();

    expect(valueDiv).toBe('11');
  });

  test("deve reducir el contador", () => {
    wrapper.find("button").at(1).simulate("click");
    const valueDiv = wrapper.find("h2").text().trim();

    expect(valueDiv).toBe('9');
  });
  test("deve resetear el contador", () => {
    wrapper.find("button").at(2).simulate("click");
    const valueDiv = wrapper.find("h2").text().trim();

    expect(valueDiv).toBe('10');
  });
});
