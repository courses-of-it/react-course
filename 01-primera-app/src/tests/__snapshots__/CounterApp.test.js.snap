// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <CounterApp /> mirar si carga el componente correactamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
